src/main/java/hudson/plugins/parameterizedtrigger/BuildTriggerConfig.java

```diff
@@ -400,18 +441,40 @@ protected Cause createUpstreamCause(AbstractBuild<?, ?> build) {
                 return new UpstreamCause((Run<?,?>)promotion.getTarget());
             }
         }
-        return new UpstreamCause((Run) build);
+        return new UpstreamCause(build);
     }
 
-    protected Future schedule(AbstractBuild<?, ?> build, AbstractProject project, int quietPeriod, List<Action> list) throws InterruptedException, IOException {
+    protected Future schedule(AbstractBuild<?, ?> build, final Job project, int quietPeriod, List<Action> list) throws InterruptedException, IOException {
+        // TODO Once it's in core and LTS is out, switch to use new ParameterizedJobMixIn convenience method
```

Comment:
Recommend specifying the since-version, `1.621`, in the comment so it is quick for a future plugin maintainer to tell whether they should use it.